web/src/pages/Route/components/DataLoader/Import.tsx

```diff
@@ -0,0 +1,262 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import React, { useState } from 'react';
+import {
+  Button,
+  Col,
+  Collapse,
+  Divider,
+  Drawer,
+  Form,
+  Input,
+  notification,
+  Result,
+  Row,
+  Select,
+  Space,
+  Upload,
+} from 'antd';
+import { UploadOutlined } from '@ant-design/icons';
+import { useIntl } from '@@/plugin-locale/localeExports';
+import OpenAPI3 from './loader/OpenAPI3';
+import type { RcFile } from 'antd/lib/upload';
+import { importRoutes } from '@/pages/Route/service';
+
+type Props = {
+  onClose?: () => void;
+  onFinish?: () => void;
+};
+
+type ImportType = 'openapi3' | 'openapi_legacy';
+type ImportState = 'import' | 'result';
+type ImportResult = {
+  success: boolean;
+  data: Record<
+    string,
+    {
+      total: number;
+      failed: number;
+      errors: string[];
+    }
+  >;
+};
+
+const entityNames = [
+  'route',
+  'upstream',
+  'service',
+  'consumer',
+  'ssl',
+  'stream_route',
+  'global_rule',
+  'plugin_config',
+  'proto',
+];
+
+const Option: React.FC<{
+  type: ImportType;
+}> = ({ type }) => {
+  switch (type) {
+    case 'openapi_legacy':
+      return <></>;
+    case 'openapi3':
+    default:
+      return <OpenAPI3 />;
+  }
+};
+
+const DataLoaderImport: React.FC<Props> = (props) => {
+  const [form] = Form.useForm();
+  const { formatMessage } = useIntl();
+  const { onClose } = props;
+  const [importType, setImportType] = useState<ImportType>('openapi3');
+  const [uploadFileList, setUploadFileList] = useState<RcFile[]>([]);
+  const [state, setState] = useState<ImportState>('import');
+  const [importResult, setImportResult] = useState<ImportResult>({
+    success: true,
+    data: {},
+  });
+
+  const onFinish = (values: Record<string, string>) => {
+    const formData = new FormData();
+    if (!uploadFileList[0]) {
+      notification.warn({
+        message: formatMessage({ id: 'page.route.button.selectFile' }),
+      });
+      return;
+    }
+    Object.keys(values).forEach((key) => {
+      formData.append(key, values[key]);
+    });
+    formData.append('file', uploadFileList[0]);
+
+    importRoutes(formData).then((r) => {
```

Comment:
don't need to handle request error cases?

Reply:
Hi, @LiteSun.

The implementation of this API does not use an HTTP status code as an error code, it will return 200 regardless of the input data, and the import error will be marked in the return `data`, while the underlying request library of the dashboard will automatically process for a 5xx error like a program error.

Reply:
It feels better to add catch to prevent application exceptions🤔

Reply:
@Baoyuantop I've added a console.error as a catch for it to help debug the error.